src/edap/edap.py

```diff
@@ -425,13 +425,130 @@ def create_all_franchises(self, source):
             self.create_franchise(code)
 
 
+class LdapTeamMixin(object):
+    """
+    Team is a posixGroup, child of organizationalUnit ou=teams that is just below the base DN.
+
+    Like division, team has machine and display names. A teams's DN begins with cn=,
+    e.g. the full team DN of a Polish publishing division is cn=PL-PUB,ou=teams,dc=entint,dc=org.
+    The description attribute of a team stores it's display name, e.g. Poland Publishing in this case.
+
+    The group's gidNumber is not important.
+    """
+
+    def get_teams(self, search=None):
+        """ Get objects (of posixGroup class) with organizational unit 'teams' by given search """
+        return self.get_objects(search=search, relative_pos='ou=teams', obj_class='posixGroup')
+
+    def get_team(self, name):
+        """
+        Get team by cname
+
+        Args:
+            name (str): team name
+
+        Returns:
+
+        """
+        return get_single_object(self.get_teams(f'cn={name}'))
+
+    def create_team(self, machine_name, display_name=None):
+        """
+        Create team
+
+        Args:
+            machine_name (str): team's cname
+            display_name (str): team's display name, stored in description
+
+        Returns:
+
+        """
+        display_name_bytes = display_name.encode('utf-8') if isinstance(display_name, str) else display_name
+        return self.create_group(name=machine_name, organizational_unit="teams", description=display_name_bytes)
+
+    def delete_team(self, machine_name):
+        """
+        Delete team by cname
+
+        Args:
+            machine_name (str): team's cname
+
+        Returns:
+        """
+        team_dn = f"cn={machine_name},{self.TEAMS_GROUP}"
+        return self.delete_object(team_dn)
+
+    def get_team_component_units(self, machine_name):
+        """
+        Get composing franchise and division from team cn
+        Args:
+            team (dict): ldap team
+
+        Returns:
+        """
+        franchise_name, division_name = machine_name.decode('utf-8').split('-', 1)
```

Comment:
The class contains the decoding logic, so an encoding logic would be nice as well (e.g. a static method `team_name(franchise, division`).

Reply:
Added methods to make team `display_name`, `machine_name`